### DISCUSSION TITLE
Is there a way to subscribe to comments without posting a comment?

### DISCUSSION THREAD

**Discussion by jimkring** [danirus/django-comments-xtd#243] [Ideas]
Hi there. I know that it's possible to get email notifications when people reply, if one has already commented. Is there a way for a user to subscribe to comments without actually commenting first?

**Comment by danirus** [comment]
Hi @jimkring, such subscription is not implemented but it definitely sounds interesting. 
Giving the chance to suscribe to comments posted to an object. 
How do you envision the workflow for the user? 

I see something like this: the user clicks on a link to "subscribe to comments". The link is visible under the main comment form. By clicking on the link a simple form is displayed with a single field to provide an email address. Once this form is posted a confirmation message with a subscription confirmation link is sent to the email address. Once the confirmation link is clicked the email address is added to a distribution list (this would be a new model). Two of the fields for this new model would be the same `content_type` and `object_id` that we find in the Comment model.

What do you think?

**Comment by jimkring** [comment]
Thanks for the response @danirus. For my use case, the work flow would be:

- User click a "Subscribe to Post" (or "Get Notifications for this Post") button.
- If the user is not signed in, we ask them to sign in (and our sign in / register process has it's own verification).
- If the user is signed in, we subscribe them.

I was thinking of hacking this by just adding an empty comment by the user and then tweaking the email notification code, so that others (who are subscribed to notifications) don't get notified about empty comments.

In the workflow you described, where people comment but are not signed in and just need to verify their email address, I think that's a perfectly good workflow.

**Reply by danirus** [reply]
Ok, that workflow would be supported too.
The difference between allowing to subscribe to both registered and non-registered users, or allowing only registered users may depend on a customizable setting.
I'm new to Discussions in GitHub. This conversation could perfectly be an issue tagged as feature. I'll try to convert it.
Thanks for this nice proposal @jimkring.

**Comment by jimkring** [comment]
You're welcome @danirus, and thanks for your good work on this useful project.

I'm new to GitHub discussions, too. I'm glad to have a chance to try it out.

**Comment by enzedonline** [comment]
Hi @danirus 

Did this feature end up in consideration? I'd really like to add post authors to the notification list automatically on publish. 

At the moment, I added a custom notify_author() to the save comment method. The problem with that is that if the page gets republished, each comment gets re-saved and the author gets a notification for every comment on that page all over again. I could create a table to keep track of comment_id and author_notified I guess, but would be good to have this instrinsic.

**Reply by danirus** [reply]
Hi @enzedonline, it's in the wishlist. 
I just added [a ticket](https://github.com/comments-ink/django-comments-ink/issues/40) to implement it in django-comments-ink. I had lost track of it.
